Index: backports-5.15.162-1/drivers/net/wireless/ath/wcn36xx/main.c
===================================================================
--- backports-5.15.162-1.orig/drivers/net/wireless/ath/wcn36xx/main.c
+++ backports-5.15.162-1/drivers/net/wireless/ath/wcn36xx/main.c
@@ -136,9 +136,7 @@ static struct ieee80211_supported_band w
 		.cap =	IEEE80211_HT_CAP_GRN_FLD |
 			IEEE80211_HT_CAP_SGI_20 |
 			IEEE80211_HT_CAP_DSSSCCK40 |
-			IEEE80211_HT_CAP_LSIG_TXOP_PROT |
-			IEEE80211_HT_CAP_SGI_40 |
-			IEEE80211_HT_CAP_SUP_WIDTH_20_40,
+			IEEE80211_HT_CAP_LSIG_TXOP_PROT,
 		.ht_supported = true,
 		.ampdu_factor = IEEE80211_HT_MAX_AMPDU_64K,
 		.ampdu_density = IEEE80211_HT_MPDU_DENSITY_16,
